cmd/kube-controller-manager/app/discovery.go

```diff
@@ -23,12 +23,14 @@ package app
 import (
 	"net/http"
 
-	"k8s.io/apimachinery/pkg/runtime/schema"
+	discoveryv1alpha1 "k8s.io/api/discovery/v1alpha1"
+	"k8s.io/klog"
 	endpointslicecontroller "k8s.io/kubernetes/pkg/controller/endpointslice"
 )
 
 func startEndpointSliceController(ctx ControllerContext) (http.Handler, bool, error) {
-	if !ctx.AvailableResources[schema.GroupVersionResource{Group: "discovery", Version: "v1alpha1", Resource: "endpointslices"}] {
+	if !ctx.AvailableResources[discoveryv1alpha1.SchemeGroupVersion.WithResource("endpointslices")] {
```

Comment:
is this an error condition? What would the user do in a case where they expect endpoint slices (say configured proxy to switch to `EndpointSlice`)?

Reply:
This could theoretically occur if the `--controllers` flag on `kube-controller-manager` was set to include `endpointslice` but the feature gate was not enabled. There are likely other edge cases I'm not thinking of. This just follows a pattern seen in a number of other startController methods, such as the [batch controller startup code](https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-controller-manager/app/batch.go#L34).